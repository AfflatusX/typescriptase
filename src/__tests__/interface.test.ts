import { interface1, interface2, interface3 } from "./examples";

test(
  "interface1",
  async (): Promise<void> => {
    expect(interface1.print({
      name: "interface1",
      path: "src/__tests__/interface.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "interface2",
  async (): Promise<void> => {
    expect(interface2.print({
      name: "interface2",
      path: "src/__tests__/interface.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "interface3",
  async (): Promise<void> => {
    expect(interface3.print({
      name: "interface3",
      path: "src/__tests__/interface.test.ts",
    }))
      .toMatchSnapshot();
  },
);
