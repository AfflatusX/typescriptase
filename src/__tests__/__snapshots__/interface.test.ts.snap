// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interface1 1`] = `
"interface If1<T> {
  kind: \\"IF1\\";
}
"
`;

exports[`interface2 1`] = `
"export interface If2<T, V> extends interface1 {
  item: string;
}
"
`;

exports[`interface3 1`] = `
"export interface If3 extends interface1, interface2 {
  test: null | string;
  test2: If2;
}
"
`;
